src/SignService/Utils/DirectoryUtility.cs

```diff
@@ -31,6 +38,23 @@ static void DeleteDirectory(string fullPath)
             }
         }
 
+        static async Task DeleteDirectoryAsync(string fullPath)
+        {
+            if (!Directory.Exists(fullPath))
```

Comment:
Is this needed? Doesnt delete noop if it doesn't exist?

Reply:
It is needed, you're thinking of CreateDirectory, that one noops

Reply:
You're right [windows](https://github.com/dotnet/corefx/blob/master/src/System.IO.FileSystem/src/System/IO/FileSystem.Windows.cs#L166-L177) and [*nix](https://github.com/dotnet/corefx/blob/master/src/System.IO.FileSystem/src/System/IO/FileSystem.Unix.cs#L208-L211) throws